src/main/java/org/spongepowered/api/entity/living/player/Player.java

```diff
@@ -259,4 +263,31 @@ default GameModeData getGameModeData() {
      */
     void setSleepingIgnored(boolean sleepingIgnored);
 
+    /**
+     * Gets a collection of {@link ServerBossBar}s this player is a member of.
+     *
+     * <p>The returned collection is unmodifiable.</p>
+     *
+     * @return The boss bars
+     */
+    Collection<ServerBossBar> getBossBars();
```

Comment:
@gabizou Probably just using an `@Inject` to the underlying NMS boss bar's `addPlayer()` and `removePlayer()` (or whatever they're called).